### PR TITLE
Remove is_direct logic from the raylet

### PR DESCRIPTION
**Pull request by edoakes**
## Why are these changes needed?

Non-direct call tasks and actors are now deprecated in both python and java.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

**Comment by edoakes** [comment]
Dependent on https://github.com/ray-project/ray/pull/7625.

**Review by ericl — COMMENTED** [review]
lgtm if tests pass. Can you trigger a full retest by touching some files in the tune and rllib directories?

File: src/ray/raylet/node_manager.cc [hunk 1]
```diff
@@ -2854,7 +2832,8 @@ void NodeManager::HandleObjectMissing(const ObjectID &object_id) {
     // So here we check for direct actor creation task explicitly to allow this case.
     auto iter = waiting_task_id_set.begin();
     while (iter != waiting_task_id_set.end()) {
-      if (IsDirectActorCreationTask(*iter)) {
+      // XXX
```
Comment (zhijunfu) [inline src/ray/raylet/node_manager.cc]: ？？

**Review by zhijunfu — APPROVED** [review]
LGTM, thanks

**Pull request merged**